Sources/Sentry/SentryAppStartTrackingIntegration.m

```diff
@@ -0,0 +1,71 @@
+#import "SentryAppStartTrackingIntegration.h"
+#import "SentryAppStartTracker.h"
+#import "SentryDefaultCurrentDateProvider.h"
+#import "SentryLog.h"
+#import <Foundation/Foundation.h>
+#import <SentryAppStateManager.h>
+#import <SentryClient+Private.h>
+#import <SentryCrashAdapter.h>
+#import <SentryDispatchQueueWrapper.h>
+#import <SentryHub.h>
+#import <SentrySDK+Private.h>
+#import <SentrySysctl.h>
+
+@interface
+SentryAppStartTrackingIntegration ()
+
+#if SENTRY_HAS_UIKIT
+@property (nonatomic, strong) SentryAppStartTracker *tracker;
+#endif
+
+@end
+
+@implementation SentryAppStartTrackingIntegration
+
+- (void)installWithOptions:(SentryOptions *)options
+{
+#if SENTRY_HAS_UIKIT
+    if (options.enableAppStartMeasuring && options.tracesSampleRate != nil &&
+        [options.tracesSampleRate doubleValue] != 0.0) {
+        SentryDefaultCurrentDateProvider *currentDateProvider =
```

Comment:
I thought we had a singleton to avoid allocating new ones for each class that depends on one, don't we?
If not, does it make sense to add it?

Reply:
Excellent point 👍. We don't have a singleton for it yet. I created an issue because this PR is already huge https://github.com/getsentry/sentry-cocoa/issues/1146.